pkg/apiserver/apiserver.go

```diff
@@ -126,16 +137,19 @@ func (c *Config) Complete(informers informers.SharedInformerFactory) completedCo
 
 func installAPIGroup(s *APIServer, c completedConfig) error {
 	cpGroup := genericapiserver.NewDefaultAPIGroupInfo(controlplane.GroupName, Scheme, metav1.ParameterCodec, Codecs)
-	cpStorage := map[string]rest.Storage{}
-	cpStorage["addressgroups"] = addressgroup.NewREST(c.extraConfig.addressGroupStore)
-	cpStorage["appliedtogroups"] = appliedtogroup.NewREST(c.extraConfig.appliedToGroupStore)
-	cpStorage["networkpolicies"] = networkpolicy.NewREST(c.extraConfig.networkPolicyStore)
-	cpGroup.VersionedResourcesStorageMap["v1beta1"] = cpStorage
+	cpv1alpha1Storage := map[string]rest.Storage{}
+	cpv1alpha1Storage["nodestatssummaries"] = nodestatssummary.NewREST(c.extraConfig.statsAggregator)
+	cpGroup.VersionedResourcesStorageMap["v1alpha1"] = cpv1alpha1Storage
+	cpv1beta1Storage := map[string]rest.Storage{}
+	cpv1beta1Storage["addressgroups"] = addressgroup.NewREST(c.extraConfig.addressGroupStore)
+	cpv1beta1Storage["appliedtogroups"] = appliedtogroup.NewREST(c.extraConfig.appliedToGroupStore)
+	cpv1beta1Storage["networkpolicies"] = networkpolicy.NewREST(c.extraConfig.networkPolicyStore)
+	cpGroup.VersionedResourcesStorageMap["v1beta1"] = cpv1beta1Storage
```

Comment:
I forgot to point it out previously, and there is no need to change anything, but isn't it a bit of an anti-pattern to introduce a v1alpha1 version of the API when we already have a v1beta1 version? I feel like the right thing to do would have been to introduce v1beta2, with support for both internal NPs and nodestatssummaries.

Reply:
Thanks Quan. It seems that both solutions are acceptable. But since we are still in pre-v1, yes it makes a bit more sense to me to add it to the existing v1beta1, and avoid the extra API version.